### ISSUE TITLE
ember new ignores --yarn

### ISSUE THREAD

**Issue by gossi** [ember-cli/ember-cli#7511]
Running `ember new foobar --yarn` ignores the `--yarn` argument. Turns out this if-block is false:

https://github.com/ember-cli/ember-cli/blob/80b4a7a5525355f405266d9660cf41915efbaa49/lib/tasks/npm-task.js#L104-L114

Also see this PR: https://github.com/ember-cli/ember-cli/pull/7481

My System:

```
$ ember version --verbose && npm --version && yarn --version
ember-cli: 2.17.1
http_parser: 2.7.0
node: 8.9.3
v8: 6.1.534.48
uv: 1.15.0
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
nghttp2: 1.25.0
openssl: 1.0.2n
icu: 60.2
unicode: 10.0
cldr: 32.0.1
tz: 2017c
os: darwin x64
5.6.0
1.3.2
```

**Comment by lennyburdette** [comment]
I see my mistake: I checked that `useYarn` is never passed to `NpmInstall#run` — and it’s never passed directly, only through the `Command#runTask` abstraction. We should probably pick one or the other: either `useYarn` is passed to the task’s constructor or it’s passed to `run`, but not both. I’m guessing we’ll need to go with the latter since the constructor isn’t available when using `runTask`. Alternatively, we could add `useYarn` to `Command#_env` so that it’s passed to every task’s constructor. 

I’m computer-less through the holidays—sorry for not being able to address this sooner!

**Comment by Turbo87** [comment]
This got fixed #7516
